simplesat/tests/test_test_utils.py

```diff
@@ -0,0 +1,123 @@
+import os.path
+import sys
+import textwrap
+
+from egginst._compat import StringIO
+from egginst.tests.common import mkdtemp
+
+from enstaller.new_solver import Requirement
+from enstaller.package import RepositoryPackageMetadata
+from enstaller.repository_info import BroodRepositoryInfo
+from enstaller.solver import JobType
+from enstaller.solver.request import _Job
+
+from okonomiyaki.platforms import PythonImplementation
+
+from ..test_utils import Scenario, parse_package_list, repository_factory
+
+if sys.version_info[0] == 2:
+    import unittest2 as unittest
+else:
+    import unittest
+
+
+P = RepositoryPackageMetadata._from_pretty_string
+
+
+class TestRepositoryFactory(unittest.TestCase):
+    def test_simple(self):
+        # Given
+        package_strings = [
+            "MKL 10.3-1",
+            "numpy 1.8.1-1; depends (MKL ~= 10.3)",
+            "numpy 1.8.1-2; depends (MKL ~= 10.3)",
+        ]
+        repository_packages = [
+            "MKL 10.3-1",
+            "numpy 1.8.1-2",
+        ]
+        repository_info = BroodRepositoryInfo("https://acme.com", "acme/loony")
+        r_numpy = P(
+            "numpy 1.8.1-2; depends (MKL ~= 10.3)", repository_info,
```

Comment:
`~=` defines a special matching comparator for which two versions are equal iff their upstream version matches. E.g. `1.9.2-1` and `1.9.2-2` match w/ `~=`, but `1.8.2-1` and `1.9.2-1` don't.